internal/kafka/internal/handlers/cluster.go

```diff
@@ -0,0 +1,90 @@
+package handlers
+
+import (
+	"net/http"
+
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/internal/kafka/internal/api/public"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/internal/kafka/internal/presenters"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/internal/kafka/internal/services"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/acl"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/api"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/client/observatorium"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/errors"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/handlers"
+)
+
+type clusterHandler struct {
+	kasFleetshardOperatorAddon services.KasFleetshardOperatorAddon
+	observabilityConfiguration *observatorium.ObservabilityConfiguration
+	clusterService             services.ClusterService
+	accessControlListConfig    *acl.AccessControlListConfig
+}
+
+func NewClusterHandler(kasFleetshardOperatorAddon services.KasFleetshardOperatorAddon, observabilityConfiguration *observatorium.ObservabilityConfiguration,
+	clusterService services.ClusterService, accessControlListConfig *acl.AccessControlListConfig) *clusterHandler {
+	return &clusterHandler{
+		kasFleetshardOperatorAddon: kasFleetshardOperatorAddon,
+		observabilityConfiguration: observabilityConfiguration,
+		clusterService:             clusterService,
+		accessControlListConfig:    accessControlListConfig,
+	}
+}
+
+func (h clusterHandler) RegisterEnterpriseCluster(w http.ResponseWriter, r *http.Request) {
+	var clusterPayload public.EnterpriseOsdClusterPayload
+
+	ctx := r.Context()
+
+	cfg := &handlers.HandlerConfig{
+		MarshalInto: &clusterPayload,
+		Validate: []handlers.Validate{
+			handlers.ValidateLength(&clusterPayload.ClusterId, "cluster id", ClusterIdLength, &ClusterIdLength),
+
+			handlers.ValidateExternalClusterId(&clusterPayload.ClusterExternalId, "external cluster id"),
+
+			handlers.ValidateClusterId(&clusterPayload.ClusterId, "cluster id"),
+
+			ValidateClusterIdIsUnique(&clusterPayload.ClusterId, h.clusterService),
+
+			handlers.ValidateDnsName(&clusterPayload.ClusterDnsName, "cluster dns name"),
+		},
+		Action: func() (interface{}, *errors.ServiceError) {
+
+			claims, claimsErr := getClaims(ctx)
+
+			if claimsErr != nil {
+				return nil, claimsErr
+			}
+
+			orgId, getOrgIdErr := claims.GetOrgId()
+
+			if getOrgIdErr != nil {
+				return nil, errors.GeneralError(getOrgIdErr.Error())
+			}
+
+			if h.accessControlListConfig.EnableAccessList {
+				isOrgAccepted := h.accessControlListConfig.AccessList.IsOrganisationAccepted(orgId)
+
+				if !isOrgAccepted {
+					return nil, errors.GeneralError("organisation '%s' is not authorized to register enterprise cluster", orgId)
+				}
+			}
```

Comment:
I am wondering how does this logic interplay with `AccessControlListMiddleware` logic: this middleware controls access to all the resources/endpoints.

I think it won't fair well. 
e.g 
1. if access control is disabled, then no access to any endpoint is granted meaning everyone except for those listed in the list, are locked out

And thus I propose we've a complete and yet "temporary" separate configuration (until we've quota in place) to control only access to the enterprise clusters registration endpoint.

Reply:
thank you!